anyway/widgets/suburban_widgets/killed_and_injured_count_per_age_group_widget.py

```diff
@@ -0,0 +1,201 @@
+from collections import defaultdict
+from datetime import datetime
+from typing import Callable, Dict, Tuple, List
+
+from flask_babel import _
+from flask_sqlalchemy import BaseQuery
+from sqlalchemy import func, asc
+
+from anyway.app_and_db import db
+from anyway.backend_constants import BE_CONST, InjurySeverity as IS
+from anyway.models import InvolvedMarkerView
+from anyway.request_params import RequestParams
+from anyway.utilities import parse_age_from_range
+from anyway.widgets.suburban_widgets.sub_urban_widget import SubUrbanWidget
+from anyway.widgets.widget import register
+from anyway.widgets.widget_utils import (
+    order_by_list_stack_bar_widget,
+    gen_entity_labels,
+    add_empty_keys_to_gen_two_level_dict,
+)
+
+# RequestParams is not hashable, so we can't use functools.lru_cache
+cache_dict = {}
+AGE_RANGE_DICT = {0: 4, 5: 14, 15: 19, 20: 24, 25: 44, 45: 64, 65: 200}
+
+
+@register
+class KilledInjuredCountPerAgeGroupStackedWidget(SubUrbanWidget):
+    name: str = "killed_and_injured_count_per_age_group_stacked"
+
+    def __init__(self, request_params: RequestParams):
+        super().__init__(request_params, type(self).name)
+        self.rank = 30
+
+    def generate_items(self) -> None:
+        raw_data = filter_and_group_injured_count_per_age_group(self.request_params)
+
+        partial_processed = add_empty_keys_to_gen_two_level_dict(
+            raw_data,
+            get_age_range_list(),
+            IS.codes(),
+        )
+
+        structured_data_list = []
+        for age_group, severity_dict in partial_processed.items():
+            series_severity = [
+                {"label_key": IS(severity_value).get_label(), "value": count}
+                for severity_value, count in severity_dict.items()
+            ]
+            structured_data_list.append({"label_key": age_group, "series": series_severity})
```

Comment:
It will be simpler to sort each list separately, instead of writing a method that gets the whole structure, and needs to loop in order to get the lists to sort.
In addition it will allow to use the list.sort() method, which will be simpler and result with less code.

Reply:
We need to sort at least 4 more widgets, so i want to create something that is generic, that way can insert one line into each widget code and get a sorted struct

Reply:
In each of these widgets we can sort the lists before they are combined into the bigger structure, isn't it?
What do we gain by doing the sort after the lists are inserted in the bigger structure?
The drawbacks in my opinion are:
1 - More code
2 - A third method that needs to be familiar with the result structure, and will need to be changed when it changes.